Foundation/NSNumber.swift

```diff
@@ -461,9 +461,12 @@ open class NSNumber : NSValue {
     }
     
     open var boolValue: Bool {
-        return int64Value != 0
+        // Darwin Foundation NSNumber appears to have a bug and return false for NSNumber(value: Int64.min).boolValue,
+        // even though the documentation says:
+        // "A 0 value always means false, and any nonzero value is interpreted as true."
+        return (int64Value != 0) && (int64Value != Int64.min)
```

Comment:
yea... I've tried to fix this a number of times... sadly it is a 32 bit truncation that occurs. Unfortunately there are a number of apps that rely on that behavior. The actual code here is effectively `int32Value != 0` but lets keep the way it should be and use the previous value